File: app/src/main/java/com/neome/api/meta/base/dto/DefnLayoutCard.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import com.neome.api.meta.base.Types.EnumDefnGridRenderingMode;
import com.neome.api.meta.base.Types.MetaIdField;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class DefnLayoutCard extends DefnLayoutGrid
{
  @Nullable
  public DefnDtoLayoutCardFilter filter;

  @Nullable
  public MetaIdField groupByFieldId;

  @Nullable
  public Boolean hideBorders;

  public DefnDtoLayoutCardItem item;

  @Nullable
  public Long numOfColumns;

  @Nullable
  public EnumDefnGridRenderingMode renderingMode;

  @Nullable
  public Boolean showSearchBar;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnLayoutCard.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.dto.DefnDtoLayoutCardFilter
import com.neome.api.meta.base.dto.DefnDtoLayoutCardItem
import com.neome.api.meta.base.dto.DefnLayoutGrid
import com.neome.api.meta.base.Types.EnumDefnGridRenderingMode
import com.neome.api.meta.base.Types.MetaIdField

interface DefnLayoutCard : DefnLayoutGrid
{
  val filter: DefnDtoLayoutCardFilter?
  val groupByFieldId: MetaIdField?
  val hideBorders: Boolean?
  val item: DefnDtoLayoutCardItem
  val numOfColumns: Long?
  val renderingMode: EnumDefnGridRenderingMode?
  val showSearchBar: Boolean?
}